crates/bevy_utils/macros/src/iterable_enum.rs

```diff
@@ -0,0 +1,63 @@
+use proc_macro::TokenStream;
+use quote::{quote, quote_spanned, ToTokens};
+use syn::{__private::Span, spanned::Spanned, DataEnum};
+
+use crate::paths;
+
+pub fn parse_iterable_enum_derive(input: TokenStream) -> TokenStream {
```

Comment:
Instead of a `get_at`, could we not just generate a `const SLICE: &[Self]` in the trait?

Reply:
Pretty smart idea. I’ll try

Reply:
Could we lose the trait altogether and just generate an associated const on the type itself?